core/Archiver/Request.php

```diff
@@ -65,12 +65,29 @@ public function getUrl()
     public function setUrl($url)
     {
         $this->url = $url;
+
+        // TODO: revisit in matomo 4
+        // period=range&date=last1/period=range&date=previous1 can cause problems during archiving due to Parameters::isDayArchive()
+        if (preg_match('/[&?]period=range/', $this->url)) {
+            $this->changeParam('period', 'day');
```

Comment:
@diosmosis not sure I understand. Wouldn't we only replace the period = range when date is last1 or previous1?

Reply:
Yes, missed that